core/shell.py

```diff
@@ -122,7 +122,7 @@ def SourceStartupFile(fd_state, rc_path, lang, parse_ctx, cmd_ev, errfmt):
   rc_line_reader = reader.FileLineReader(f, arena)
   rc_c_parser = parse_ctx.MakeOshParser(rc_line_reader)
 
-  with alloc.ctx_Location(arena, source.SourcedFile(rc_path)):
+  with alloc.ctx_Location(arena, source.SourcedFile(rc_path, -1)):
```

Comment:
The -1 could be runtime.NO_SPID

(the location is supposed to be where the `source` statement happens, but for the rcfile there's no `source`.  Might want to polish this later)